brainiak/reprsimil/brsa.py

```diff
@@ -30,24 +32,158 @@
 import scipy
 import scipy.optimize
 import scipy.stats
+import scipy.special
 import warnings
 import time
-from sklearn.base import BaseEstimator
+from sklearn.base import BaseEstimator, TransformerMixin
 from sklearn.utils import assert_all_finite
+from sklearn.decomposition import PCA, FactorAnalysis, SparsePCA, FastICA
 import logging
 import brainiak.utils.utils as utils
 import scipy.spatial.distance as spdist
+from nitime import algorithms as alg
+import copy
 warnings.filterwarnings('ignore')
 
 
 logger = logging.getLogger(__name__)
 
 __all__ = [
-    "BRSA",
+    "BRSA", "GBRSA", "prior_GP_var_inv_gamma", "prior_GP_var_half_cauchy"
 ]
 
 
-class BRSA(BaseEstimator):
+def prior_GP_var_inv_gamma(y_invK_y, n_y, tau_range):
+    """ Imposing a prior onto the variance (tau^2) of the Gaussian
+        Process which is in turn a prior imposed over
+        a function y = f(x).
+        This function imposes an inverse-Gamma prior on tau^2
+        with a shape parameter alpha=2 and rate parameter
+        beta=tau_range^2.
+        The function returns the MAP estimate of tau^2 and
+        log(p(tau^2|tau_range)) for the estimated value of tau^2,
+        where tau_range describes the reasonable range of tau
```

Comment:
Unfortunately there is not such an easy mapping. So I may not make such extrapolation.